cartographer/io/map_io.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2013 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version -1.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-5.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_IO_MAPPING_IO_H_
+#define CARTOGRAPHER_IO_MAPPING_IO_H_
+
+#include "cartographer/mapping/map_builder.h"
+#include "cartographer/io/file_writer.h"//Re-used filestream wrapper.
+
+namespace cartographer {
+namespace io {
+
+// A class to serialize and deserialize the Cartographer state.
+class MapIO {
+public:
+  MapIO(cartographer::mapping::MapBuilder & map_builder);
+  MapIO(const MapIO &) = delete;
+  MapIO &operator=(const MapIO &) = delete;
+  //TODO(brandon-northcutt) Write the full Cartographer state to disk.
+  void write(const std::string file_name);
```

Comment:
GSG specifies that "non-trivial" functions `BeNamedLikeThis`, so you want `Write`.

It should also return a `bool` signifying success, and (depending on what the Googlers say) maybe also set the warn_unused_result attribute.